cmd/crank/foundation/resource/xrm/client.go

```diff
@@ -115,34 +117,58 @@ func getResourceChildrenRefs(r *resource.Resource, getConnectionSecrets bool) []
 	if xrcNamespace := obj.GetNamespace(); xrcNamespace != "" {
 		// This is an XRC, get the XR ref, we leave the connection secret
 		// handling to the XR
-		xrc := claim.Unstructured{Unstructured: obj}
-		if ref := xrc.GetResourceReference(); ref != nil {
-			refs = append(refs, v1.ObjectReference{
-				APIVersion: ref.APIVersion,
-				Kind:       ref.Kind,
-				Name:       ref.Name,
-			})
-		}
-
-		if getConnectionSecrets {
-			xrcSecretRef := xrc.GetWriteConnectionSecretToReference()
-			if xrcSecretRef != nil {
-				ref := v1.ObjectReference{
-					APIVersion: "v1",
-					Kind:       "Secret",
-					Name:       xrcSecretRef.Name,
-					Namespace:  xrcNamespace,
+		out := &v1.ObjectReference{}
+		// Determine if this is a namespaced XR or an XRC
+		// XRCs will have the spec.crossplane.resourceRef or spec.resourceRef field
+		// If either field is found, we treat it as an XRC
+		checkNewField := fieldpath.Pave(obj.Object).GetValueInto("spec.crossplane.resourceRef", out)
+		checkLegacyField := fieldpath.Pave(obj.Object).GetValueInto("spec.resourceRef", out)
+		if checkNewField == nil || checkLegacyField == nil {
+			xrc := claim.Unstructured{Unstructured: obj}
+			if ref := xrc.GetResourceReference(); ref != nil {
+				refs = append(refs, v1.ObjectReference{
+					APIVersion: ref.APIVersion,
+					Kind:       ref.Kind,
+					Name:       ref.Name,
+				})
+			}
```

Comment:
this is odd - it seems to be the exact same changes that I reviewed earlier today (and think we should do differently) from @man-ish-k in https://github.com/crossplane/crossplane/pull/6549/files - how did this code end up in your PR too? it's from another contributor/fork, so I can't immediately blame a bad merge. Do you know how this happened? 🤔

Reply:
I fixed it first, made the change independently -- this PR has been open a long time.  If it's on main now, it probably just needs a rebase?

Reply:
It looks like @man-ish-k made the change in an earlier PR before this one in https://github.com/crossplane/crossplane/pull/6380, then closed that to open https://github.com/crossplane/crossplane/pull/6549 that was merged recently.

Do you all work together or know each other? I'm still not understanding how this duplication happened exactly and I definitely would like to 🙏